components/object-property-list/object-property-list.js

```diff
@@ -30,16 +30,23 @@ class ObjectPropertyList extends LocalizeCoreElement(SkeletonMixin(LitElement))
 			::slotted(:last-child), slot :last-child {
 				--d2l-object-property-list-item-separator-display: none;
 			}
+			::slotted([slot="status"]) {
+				margin-inline-end: 0.45rem;
+			}
 		`];
 	}
 
 	render() {
-
 		const slotContents = this.skeleton && this.skeletonCount > 0 ? [...Array(this.skeletonCount)].map(() => html`
 			<d2l-object-property-list-item text="${this.localize('components.object-property-list.item-placeholder-text')}" skeleton></d2l-object-property-list-item>
 		`) : nothing;
 
-		return html`<slot class="d2l-body-small">${slotContents}</slot>`;
+		return html`
+			<div class="d2l-body-small">
+				<slot name="status"></slot>
```

Comment:
Wondering about how this new slot should behave when in skeleton mode... should it maybe be included in the `skeletonCount` behaviour? Or were you thinking they'd just put `<d2l-status-indicator skeleton>` inside the slot? (although status indicators don't appear to support skeletons currently)

Reply:
Yeah, `skeletonCount` requires there to be nothing slotted at all, and I don't think it makes sense to add an extra attribute on `object-property-list` to indicate that there should be an additional skeletonized status indicator. I was thinking `<d2l-status-indicator skeleton>` but I did notice that it doesn't support skeleton yet.

Reply:
I am wondering if we should have this new slot or not. The slot very loosely enables us to control where the status indicator is rendered. Could we achieve the same thing by using some CSS like this? Of course, it doesn't explicitly require that the user place it first, but technically, we're not really limiting where they can render this anyway.

```css
::slotted(d2l-status-indicator) {
     margin-inline-end: 0.45rem;
}
```

The other thought that came to mind was to define a `d2l-object-property-list-item`, but that seems like overkill.

Reply:
The slot seemed appropriate to me because:

1. It limits it to only one status indicator, and forces it to render first (both are design requirements)
2. It makes the use case/support of status indicators clear and explicit, without having to create a new web component for it
3. It seems conceptually like a good match for the purpose of named slots

Yep we could achieve the margin without the slot, just wondering if there's any downside to using a slot though?

Reply:
Ah cool, yep I wasn't sure what happened when putting two items into a single named slot. I guess the good news is, if we ever need multiple status indicators at the start it's sort of supported already?

Putting a status indicator without a slot would be unsupported behaviour, which I called out in the readme "Best practices". But I think your screenshot is still a win - the slotted ones have the right margin and no separator, and the one used incorrectly looks as good as possible, with a separator and no extra right margin.

Reply:
One downside I just discovered - there's no screen-reader pause between reading a status and reading the first item. Not sure if that's required - I think the only way to get that behaviour would be to make a custom web-component for `d2l-object-property-list-item-status` 🤦.

Reply:
Cool. If we do want to lock things down... we've done this sort of thing in one or two places:
```css
::slotted([slot="status"]) {
	display: none;
}
::slotted(d2l-status-indicator[slot="status"]) {
	display: inline-block;
	margin-inline-end: 0.45rem;
}
```
That might be desirable, since the slot isn't intended to be general purpose. Technically we could target `d2l-status-indicator` in the default slot and hide them, if we really do want to enforce not rendering them there.